src/dioptra/restapi/v1/schemas.py

```diff
@@ -24,63 +24,110 @@
 from dioptra.restapi.v1.users.schema import UserRefSchema
 
 
-def generate_base_resource_schema(name: str) -> type[Schema]:
+def generate_base_resource_schema(name: str, snapshot: bool) -> type[Schema]:
```

Comment:
Given that Tags are a somewhat special object, I'm wondering if there should be a generic generate_base_schema function to handle them as well, or if they should just be handled as a special case alone.  
Tags require   
- id: int,
- user: UserRef,
- group: GroupRef,
- created_on: Timestamp,
- last_modified_on: Timestamp,


A potential pattern for the builder could a series of functions which append certain keys:values to a base dict depending on the resource. That is pulling out the functionality of the snapshot if statement into it's own function and creating another one corresponding to resource, tag, snapshot etc. 

As it stands, I can see it overcomplicating it for no real benefit. Maybe as an alternative we can just add to the function's docstring something like `Builder function for all resource snapshot and resource base schemas. Tags are a separate entity type and are handled in its respective schema.py file.`

Reply:
See #399 for Tags schemas.

Reply:
keeping this as is for now. We can revisit if we find that there are more "special cases" that could be handled in a more general way.